nixos/modules/programs/tsm-client.nix

```diff
@@ -243,7 +259,12 @@ let
   makeDsmSysStanza = {servername, ... }@serverCfg:
     let
       # drop special values that should not go into server config block
-      attrs = removeAttrs serverCfg [ "servername" "genPasswd" ];
+      attrs = removeAttrs serverCfg [ "servername" "genPasswd"
+        # XXX migration code for freeform settings, these can be removed after 23.11:
+        "assertions" "warnings"
+        "extraConfig" "text"
+        "name" "server" "port" "node" "passwdDir" "includeExclude"
```

Comment:
Did you consider adding a `settings` option on the server submodule, rather than commingling the freeform options with the existing ones? It might eliminate the need to ignore options here and simplify logic in a few places.

Reply:
Yes, but I don't think it's worth the effort: Most attribute keys listed here are due to the migration mechanism; as soon as the migration code gets removed, there are only two attributes left.  The `servername` is usually defined with the attribute key below `servers`.  So we would add another attribute set level only for a single option.